pkg/manager/manager_test.go

```diff
@@ -302,6 +302,24 @@ var _ = Describe("manger.Manager", func() {
 				Expect(err).To(HaveOccurred())
 				Expect(err.Error()).To(ContainSubstring("unable to find leader election namespace: not running in-cluster, please specify LeaderElectionNamespace"))
 			})
+
+			// We must keep this default until we are sure all controller-runtime users have upgraded from the original default
+			// ConfigMap lock to a controller-runtime version that has this new default. Many users of controller-runtime skip
```

Comment:
How are we going to track this for all our users? We could maybe say that v0.7.x is going to be the last release to default to ConfigMap locks, and going forward there is going to be a breaking change that user need to take into account. 

The change can be part of upgrading and should be very explicit in release notes.

Reply:
The `MultiLock` we set up requires both a `ConfigMap` and a `Lease`. It is intended for transitions. I would like to start using this ASAP (so in v0.7, not later), because we will have to stick to this for a very long time, until we are sure everyone had a c-r release that had the `MultiLock`. Othewise we can break users in a subtile way if their controller suddenly used a `Lease` but not a `ConfigMap` anymore, resulting in two leaders existing at the same time.

We also want to allow ppl to explicitly configure a different strategy if they know that all their users had a release with the `MultiLock`. However the default here has to be like this for a long time (certainly more than release, I would say at least a year after the first c-r release that has this change).